Trading/Mode/grid_trading_mode/tests/test_grid_trading_mode.py

```diff
@@ -371,8 +359,265 @@ async def test_create_orders_with_fixed_volume_per_order():
+        _check_created_orders(producer, trading_api.get_open_orders(exchange_manager), 100)
+
+
+async def test_start_after_offline_full_sell_side_filled_orders_with_recent_trades():
+    symbol = "BTC/USDT"
+    async with _get_tools(symbol) as (producer, _, exchange_manager):
+        # first start: setup orders
+        producer.sell_funds = decimal.Decimal("1")  # 20 orders
+        producer.buy_funds = decimal.Decimal("1")  # 24 orders
+        orders_count = 20 + 24
+
+        price = 100
+        trading_api.force_set_mark_price(exchange_manager, producer.symbol, price)
+        await producer._ensure_staggered_orders()
+        await asyncio.create_task(_check_open_orders_count(exchange_manager, orders_count))
+        original_orders = copy.copy(trading_api.get_open_orders(exchange_manager))
+        assert len(original_orders) == orders_count
+        pre_portfolio = trading_api.get_portfolio_currency(exchange_manager, "USDT").available
+
+        # offline simulation: orders get filled but not replaced => price got up to more than the max price
+        open_orders = trading_api.get_open_orders(exchange_manager)
+        offline_filled = [o for o in open_orders if o.side == trading_enums.TradeOrderSide.SELL]
+        for order in offline_filled:
+            await _fill_order(order, exchange_manager, trigger_update_callback=False, producer=producer)
+        post_portfolio = trading_api.get_portfolio_currency(exchange_manager, "USDT").available
+        assert pre_portfolio < post_portfolio
+        assert len(trading_api.get_open_orders(exchange_manager)) == orders_count - len(offline_filled)
+
+        # back online: restore orders according to current price
+        price = max(order.origin_price for order in offline_filled) * 2
+        trading_api.force_set_mark_price(exchange_manager, producer.symbol, price)
+        await producer._ensure_staggered_orders()
+        # restored orders (and create up to 50 orders as all orders can be created)
+        assert producer.operational_depth > orders_count
+        await asyncio.create_task(_check_open_orders_count(exchange_manager, producer.operational_depth))
+        assert 0 <= trading_api.get_portfolio_currency(exchange_manager, "USDT").available <= post_portfolio
+        assert 0 <= trading_api.get_portfolio_currency(exchange_manager, "BTC").available
+        open_orders = trading_api.get_open_orders(exchange_manager)
+        assert all(
+            order.side == trading_enums.TradeOrderSide.BUY
+            for order in open_orders
+        )
+        _check_created_orders(producer, trading_api.get_open_orders(exchange_manager), 100)
+
+
+async def test_start_after_offline_full_sell_side_filled_orders_price_back():
+    symbol = "BTC/USDT"
+    async with _get_tools(symbol) as (producer, _, exchange_manager):
+        # first start: setup orders
+        producer.sell_funds = decimal.Decimal("1")  # 20 orders
+        producer.buy_funds = decimal.Decimal("1")  # 24 orders
```

Comment:
What does `1` means here?

Reply:
1 BTC, it's kinda random, the idea is to allow as many orders as possible

Reply:
Oh okay, I didn't expect this to be a quantity of BTC 😄